tests/src/python/test_provider_afs.py

```diff
@@ -1223,6 +1225,178 @@ def testRenderer(self):
+                    61,
+                    51,
+                    255
+                  ],
+                  "label": null
+                },
+                {
+                  "value": 2500000,
+                  "color": [
+                    123,
+                    66,
+                    56,
+                    255
+                  ],
+                  "label": "> 2,500,000"
+                }
+              ]
+            },
+            {
+              "type": "sizeInfo",
+              "target": "outline",
+              "expression": "view.scale",
+              "valueExpression": "$view.scale",
+              "stops": [
+                {
+                  "size": 1.5,
+                  "value": 3468153
+                },
+                {
+                  "size": 0.75,
+                  "value": 10837979
+                },
+                {
+                  "size": 0.375,
+                  "value": 43351915
+                },
+                {
+                  "size": 0,
+                  "value": 86703831
+                }
+              ]
+            }
+          ],
+          "authoringInfo": {
+            "visualVariables": [
+              {
+                "type": "colorInfo",
+                "minSliderValue": 10151,
+                "maxSliderValue": 15185477,
+                "theme": "high-to-low"
+              }
+            ]
+          },
+          "type": "classBreaks",
+          "field": "SUM",
+          "minValue": -9007199254740991,
+          "classBreakInfos": [
+            {
+              "symbol": {
+                "color": [
+                  170,
+                  170,
+                  170,
+                  255
+                ],
+                "outline": {
+                  "color": [
+                    194,
+                    194,
+                    194,
+                    64
+                  ],
+                  "width": 0.75,
+                  "type": "esriSLS",
+                  "style": "esriSLSSolid"
+                },
+                "type": "esriSFS",
+                "style": "esriSFSSolid"
+              },
+              "classMaxValue": 9007199254740991
+            }
+          ]
+        },
+        "transparency": 20
+      },
+      "allowGeometryUpdates": true
+    }""")
+
+        with open(sanitize(endpoint, '/query?f=json_where=1=1&returnIdsOnly=true'), 'wb') as f:
+            f.write(b"""
+        {
+         "objectIdFieldName": "OBJECTID",
+         "objectIds": [
+          1
+         ]
+        }
+        """)
+
+        # Create test layer
+        vl = QgsVectorLayer("url='http://" + endpoint + "' crs='epsg:3857'", 'test', 'arcgisfeatureserver')
+        self.assertTrue(vl.isValid())
+        self.assertIsNotNone(vl.dataProvider().createRenderer())
+        self.assertIsInstance(vl.renderer(), QgsGraduatedSymbolRenderer)
+        self.assertIsInstance(vl.renderer().sourceSymbol(), QgsSymbol)
```

Comment:
Can you also test for the ranges created here? Eg check the count of ranges for the renderer, and their min/max values?

Reply:
Done